src/Azure/Azure.Quantum.Client/JobManagement/CloudJob.cs

```diff
@@ -38,45 +40,73 @@ public CloudJob(IWorkspace workspace, JobDetails jobDetails)
         /// <summary>
         /// Gets the job id.
         /// </summary>
-        public string Id => Details.Id;
+        public virtual string Id => Details.Id;
+
+        /// <summary>
+        /// Gets the job id.
+        /// </summary>
+        public virtual string Name => Details.Name;
 
         /// <summary>
         /// Gets whether the job execution has completed.
         /// </summary>
-        public bool InProgress => Status != "Cancelled" &&
+        public virtual bool InProgress => Status != "Cancelled" &&
                                   Status != "Failed" &&
                                   Status != "Succeeded";
 
         /// <summary>
         /// Gets the status of the submitted job.
         /// </summary>
-        public string Status => Details.Status;
+        public virtual string Status => Details.Status?.ToString();
 
         /// <summary>
         /// Gets whether the job execution completed successfully.
         /// </summary>
-        public bool Succeeded => Status == "Succeeded";
+        public virtual bool Succeeded => Status == "Succeeded";
 
         /// <summary>
         /// Gets an URI to access the job.
         /// </summary>
-        public Uri Uri => GenerateUri();
+        public virtual Uri Uri => GenerateUri();
 
         /// <summary>
         /// Gets the workspace.
         /// </summary>
-        public IWorkspace Workspace { get; private set; }
+        public virtual IWorkspace Workspace { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the unique identifier for the provider.
+        /// </summary>
+        public virtual string ProviderId => this.Details?.ProviderId;
+
+        /// <summary>
+        /// Gets or sets the target identifier to run the job.
```

Comment:
I only see an accessor for this and `ProviderId`, but the doc comment seems to indicate a mutator as well?